service/frontend/workflow_handler.go

```diff
@@ -3829,27 +3841,66 @@ func (wh *WorkflowHandler) validateBuildIdCompatibilityUpdate(
 	return nil
 }
 
-func (wh *WorkflowHandler) historyArchived(ctx context.Context, request *workflowservice.GetWorkflowExecutionHistoryRequest, namespaceID namespace.ID) bool {
-	if request.GetExecution() == nil || request.GetExecution().GetRunId() == "" {
+func (wh *WorkflowHandler) historyArchived(ctx context.Context, execution *commonpb.WorkflowExecution, namespaceID namespace.ID) bool {
+	return wh.isWorkflowArchived(ctx, execution, namespaceID)
+}
+func (wh *WorkflowHandler) isWorkflowArchived(ctx context.Context, execution *commonpb.WorkflowExecution, namespaceID namespace.ID) bool {
```

Comment:
The naming convention seems to be `workflowArchived` without `is`. To be consistent with the argument type, it should be `executionArchived` that can be called for checking either workflow or history execution.